136 8 documentation + token generator testing code

frimic-solarly opened the pull request.

dmohns commented on docs/index.md:
```diff
@@ -34,3 +34,27 @@
 This repository contains manufacturing details as related to smart energy meter, details such as hardware design, web software, firmware, exterior casing STL files, rationale behind design, how-to document to guide a skilled user to replicate every part of the smart meter available in the repository.
 
 ![OpenSmartMeter 3D Render](assets/hardware/render_3d/opensmartmeter_render_3d_high_quality.png)
+
+## Getting Started
+
+You have now an OpenSmartMeter in your hands and you want to test it? Here is how.
+
+### Initial configuration
+ToDo: Do a tutorial on how to access the technical mode via the keypad of the OpenSmartMeter. Make pictures of the OSM.
+
+ToDo: How to set up secret key, starting code, etc.
+
+### Testing OpenPAYGO tokens
+After plugging the OSM, it will appear the following screen:
+ToDo: Provide picture
+
+By default, the OSM is deactivated, which means it doesn't provide any energy on its output.
+If the initial default configuration has been applied, you can test tokens available on the Excel sheet "Token examples.xlsx" (Find it here: OpenSmartMeter\firmware\test).
+If you want to create other token:
+1) Open the repository on an IDE with python3.9 or above.
+2) Install the python requirements (in OpenSmartMeter/firmware/test):
+   pip3 install -r requirements.txt
+3) Change initial parameters in "test_openpaygo.py"
+Token_value can be days or kWh, depending on the initial default configuration you have defined on the OSM itself.
+4) Run script and get your desired token in console output.
+
```
Could you move these to a new section in the documentation? For example a quick-start guide, here is an example in our AirLink project 
<img width="789" alt="image" src="https://github.com/EnAccess/OpenSmartMeter/assets/14202480/c5810cf0-2fa7-4470-b874-78693acd46bf">

frimic-solarly replied:
Done ;)

dmohns reviewed (commented):
Few more comments:

- Can you rebase/merge latest `main` branch? I added some changes to the linter configuration
- Can you format your Python code using [Black](https://github.com/psf/black)?
- Would it be possible to have a CSV instad of XLSX file?

frimic-solarly commented:
> Few more comments:
> 
>     * Can you rebase/merge latest `main` branch? I added some changes to the linter configuration
> 
>     * Can you format your Python code using [Black](https://github.com/psf/black)?
> 
>     * Would it be possible to have a CSV instad of XLSX file?

Hello @dmohns, 
I have rebased the branch and reformatted with Black. 
However for your third point, I think having a CSV instead of XLSX will not look good. Maybe just a .txt file or I move it to the documentation?

dmohns commented:
Great thanks, can you now fix the remaining lint errors?

> I think having a CSV instead of XLSX will not look good. 

What do you mean by will not look good? Github automatically renders CSV file, for example: https://github.com/EnAccess/OpenPAYGO-Token/blob/main/documentation/examples/example_device_list.csv

dmohns commented on docs/quick-start-guide.md:
```diff
@@ -0,0 +1,24 @@
+
+By default, the OSM is deactivated, which means it doesn't provide any energy on its output.
+If the initial default configuration has been applied, you can test tokens available on the Excel sheet "Token examples.xlsx" (Find it here: OpenSmartMeter\firmware\test).
+If you want to create other token:
+
+1. Open the repository on an IDE with python3.9 or above.
+2. Install the python requirements (in OpenSmartMeter/firmware/test):
+   pip3 install -r requirements.txt
+3. Change initial parameters in "test_openpaygo.py"
+Token_value can be days or kWh, depending on the initial default configuration you have defined on the OSM itself.
+4. Run script and get your desired token in console output.
```
As discussed, maybe here you can just add the list of pre-generated tokens that user can use immediately to test the device.

The selected lines can probably go into `README.md` in the `firmware/test` directory (need to create it 😉 ), which explains how to run the test Python.
